### PR TITLE
FEATURE/GRID-292: Make `halign` a "render property" with default of 'center'

### PR DESCRIPTION
**Pull request by Dwaynekj**
@joneit @rdepena 

API Call to test
grid.behavior.setColumnProperties(1,{halign: 'right'});
Note there is no exemption for "headers" they are treated like regular cells. If there areå semantics needed to have separate alignment that can be specified from the application layer by `getCell`

Best to test on the non-pink demo like
`demo/example.html`

as the main demo has some overrides in `getCell` that sets a different alignment

### CODE REVIEW

File: src/defaults.js [hunk 1]
```diff
@@ -364,6 +364,12 @@ module.exports = {
      * @instance
      */
     fixedColAlign: 'center',
+    /**
+     * @default
+     * @type {string}
+     * @instance
+     */
+    defaultColAlign: 'center',
```
Comment (joneit) [inline src/defaults.js]: I would go with simply `halign` but I see you're trying to parallel `fixedColAlign` so I could live with `colAlign` or `dataColAlign`... but really I prefer `halign` which is already a familiar prop in getCell config.

File: src/behaviors/Column.js [hunk 2]
```diff
@@ -9,7 +9,8 @@ var propertyNames = [
     'name',
     'header',
     'calculator',
-    'type'
+    'type',
+    'halign'
```
Comment (joneit) [inline src/behaviors/Column.js]: Not here. Rather on the column properties object, which automatically defers to the grid properties object before deferring to the theme properties object before deferring (finally) to the default properties object.

File: src/behaviors/JSON.js [hunk 3]
```diff
@@ -226,11 +227,16 @@ var JSON = Local.extend('behaviors.JSON', {
...
-        if (x === 0 && this.hasHierarchyColumn()) {
-            return 'left';
+        var align = this.getPrivateState().defaultColAlign;
+        if (x === -1) {
+            align = 'right';
+        } else if (x === 0 && this.hasHierarchyColumn()) {
+            align = 'left';
         } else {
-            return 'center';
+            align  = this.getActiveColumn(x).halign || align;
```
Comment (joneit) [inline src/behaviors/JSON.js]: should be:
```javascript
align = this.getColumnProperties(x).halign;
```
and remove line 230's definition (keep declaration)

**Comment by Dwaynekj** [comment]
Will make changes today

**Comment by Dwaynekj** [comment]
@joneit @rdepena Made changes

**Comment by Dwaynekj** [comment]
@joneit Ready for review

File: src/defaults.js [hunk 4]
```diff
@@ -369,7 +369,7 @@ module.exports = {
      * @type {string}
      * @instance
      */
-    defaultColAlign: 'center',
+    colHAlign: 'center',
```
Comment (joneit) [inline src/defaults.js]: Remove "col" because it can also pertain the the grid as a whole.

File: src/behaviors/Behavior.js [hunk 5]
```diff
@@ -185,6 +185,9 @@ var Behavior = Base.extend('Behavior', {
         var tableState = this.getPrivateState();
         var properties = Object.create(tableState);
 
+
+        properties.halign = tableState.colHAlign;
```
Comment (joneit) [inline src/behaviors/Behavior.js]: Don't do this. Just make the entry in Defaults.js "halign" and you're done.

File: src/behaviors/JSON.js [hunk 6]
```diff
@@ -60,7 +60,8 @@ var JSON = Local.extend('behaviors.JSON', {
             var column = this.addColumn({
                 index: index,
                 header: header,
-                calculator: calculator
+                calculator: calculator,
+                halign: ''
```
Comment (joneit) [inline src/behaviors/JSON.js]: Remove this line. This is not needed here and is not being used.

File: src/defaults.js [hunk 7]
```diff
@@ -364,6 +364,12 @@ module.exports = {
      * @instance
      */
     fixedColAlign: 'center',
+    /**
+     * @default
+     * @type {string}
+     * @instance
+     */
+    colHAlign: 'center',
```
Comment (joneit) [inline src/defaults.js]: Remove "col" because it can also pertain the the grid as a whole. Just make it 'halign'.

Comment (Dwaynekj) [inline src/defaults.js]: I think this should stay. We should have a col and rowAlign

**Pull request merged**